pkg/system/phase4_configuring.go

```diff
@@ -597,6 +612,78 @@ func (r *Reconciler) prepareAzureBackingStore() error {
 	return nil
 }
 
+func (r *Reconciler) prepareGCPBackingStore() error {
+
+	cloudCredsSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.GCPCloudCreds.Spec.SecretRef.Name,
+			Namespace: r.GCPCloudCreds.Spec.SecretRef.Namespace,
+		},
+	}
+
+	util.KubeCheck(cloudCredsSecret)
+	if cloudCredsSecret.UID == "" {
+		// TODO: we need to figure out why secret is not created, and react accordingly
+		// e.g. maybe we are running on AWS but our CredentialsRequest is for GCP
+		r.Logger.Infof("Secret %q was not created yet by cloud-credentials operator. retry on next reconcile..", r.GCPCloudCreds.Spec.SecretRef.Name)
+		return fmt.Errorf("cloud credentials secret %q is not ready yet", r.GCPCloudCreds.Spec.SecretRef.Name)
+	}
+	r.Logger.Infof("Secret %s was created successfully by cloud-credentials operator", r.GCPCloudCreds.Spec.SecretRef.Name)
+
+	util.KubeCheck(r.GCPContainerCreds)
+	if r.GCPContainerCreds.UID == "" {
+		r.GCPContainerCreds.StringData = cloudCredsSecret.StringData
+		r.Own(r.GCPContainerCreds)
+		if err := r.Client.Create(r.Ctx, r.GCPContainerCreds); err != nil {
+			return fmt.Errorf("got error on GCPContainerCreds creation. error: %v", err)
+		}
+	}
+	authJSON := &gcpAuthJSON{}
+	err := json.Unmarshal([]byte(cloudCredsSecret.StringData["service_account.json"]), authJSON)
+	if err != nil {
+		fmt.Println("Failed to parse secret", err)
+	}
+	projectID := authJSON.ProjectID
+	r.GCPContainerCreds.StringData["GoogleServiceAccountPrivateKeyJson"] = cloudCredsSecret.StringData["service_account.json"]
+	ctx := context.Background()
+	gcpclient, err := storage.NewClient(ctx, option.WithCredentialsJSON([]byte(cloudCredsSecret.StringData["service_account.json"])))
+	if err != nil {
+		r.Logger.Info(err)
+	}
+
+	var bucketName = strings.ToLower(randname.GenerateWithPrefix("noobaaBucket", 5))
+	if err := r.createGCPBucketForBackingStore(gcpclient, projectID, bucketName); err != nil {
+		r.Logger.Info(err)
+	}
+
+	if errUpdate := r.Client.Update(r.Ctx, r.GCPContainerCreds); errUpdate != nil {
+		return fmt.Errorf("got error on GCPContainerCreds update. error: %v", errUpdate)
+	}
+	// create backing store
+	r.DefaultBackingStore.Spec.Type = nbv1.StoreTypeGoogleCloudStorage
+	r.DefaultBackingStore.Spec.GoogleCloudStorage = &nbv1.GoogleCloudStorageSpec{
+		TargetBucket: bucketName,
+		Secret: corev1.SecretReference{
+			Name:      r.GCPContainerCreds.Name,
+			Namespace: r.GCPContainerCreds.Namespace,
+		},
+	}
+	return nil
+}
+
+func (r *Reconciler) createGCPBucketForBackingStore(client *storage.Client, projectID, bucketName string) error {
+	// [START create_bucket]
+	ctx := context.Background()
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
```

Comment:
Not sure if a timeout of 10 seconds is safe on slow connections.
Maybe raise it a bit? Like 30 seconds or so?
What do you think?
@rajatsing @nimrod-becker

Reply:
Yes, this can be increased but, in our tests, we never got timed out, we were not on very fast internet connection though.

Reply:
Lets do the same as other backing stores

Reply:
I changed it to 30 seconds.

Reply:
I changed it to 30 seconds